server/telemetry/model.go

```diff
@@ -0,0 +1,26 @@
+package telemetry
+
+import (
+	"server/model"
+
+	"gorm.io/gorm"
+)
+
+type Telemetry struct {
+	model.BaseModel
+	MetricName  DeploymentMetric `gorm:"type:string"`
+	MetricValue string
+	Step        string
+}
+
+// Getter Setters for each DeploymentMetric
+func SetMetric(db *gorm.DB, telemetry *Telemetry, metric DeploymentMetric, status string) {
+	telemetry.MetricName = metric
+	telemetry.MetricValue = status
+	db.Save(&telemetry)
+}
+
+func Metric(db *gorm.DB, metric DeploymentMetric) *gorm.DB {
```

Comment:
GetMetric would be better.  It looks like a class or object name.  Also the getter/setter should be included in the unit test in addition to or instead of getting the data directly from the DB.  I was going to point out the sonar analysis showing code coverage by unit tests, but apparently it doesn't run on pull requests.

Reply:
I just followed the Golang conventions for naming Setter/Getters (https://go.dev/doc/effective_go#Getters) :smile: 
Updated to add unit tests for getter and setter

Reply:
No problem!